### PR TITLE
[Model Monitoring] Init endpoint on the server side

### PR DESCRIPTION
**Pull request by davesh0812**
implemented https://iguazio.atlassian.net/browse/ML-7930

Fix also https://iguazio.atlassian.net/browse/ML-8826, https://iguazio.atlassian.net/browse/ML-8771

### CODE REVIEW

File: mlrun/serving/v2_serving.py [hunk 1]
```diff
@@ -116,6 +115,7 @@ def predict(self, request):
             self.ready = True
         self._versioned_model_name = None
```
Comment (Eyal-Danieli) [inline mlrun/serving/v2_serving.py]: not in use anymore
```suggestion

```

File: server/py/services/api/crud/model_monitoring/deployment.py [hunk 2]
```diff
@@ -1207,6 +1209,253 @@ def _is_the_same_cred(self, stream_path: str, tsdb_connection: str) -> bool:
...
+                        creation_strategy=creation_strategy,
+                        model_path=model_path,
+                    )
+                )
+            )
+        created_model_endpoint.extend(await asyncio.gather(*router_tasks))
+
+        return created_model_endpoint
+
+    def _extract_model_endpoints_from_function_graph(
```
Comment (Eyal-Danieli) [inline server/py/services/api/crud/model_monitoring/deployment.py:1304]: I suggest to add few lines for the func desc, in particular about the response (I have no idea what is the last str value represents)

File: server/py/services/api/crud/model_monitoring/deployment.py [hunk 3]
```diff
@@ -1207,6 +1209,253 @@ def _is_the_same_cred(self, stream_path: str, tsdb_connection: str) -> bool:
             return False
         return True
 
+    async def create_model_endpoints(self, function: dict, function_name: str):
```
Comment (Eyal-Danieli) [inline server/py/services/api/crud/model_monitoring/deployment.py:1212]: I suggest to add short description about the func logic

File: server/py/services/api/crud/model_monitoring/deployment.py [hunk 4]
```diff
@@ -1207,6 +1209,253 @@ def _is_the_same_cred(self, stream_path: str, tsdb_connection: str) -> bool:
...
+                            ),
+                            step.model_endpoint_creation_strategy,
+                            step.class_args.get("model_path", ""),
+                        )
+                    )
+        return model_endpoints_instructions
+
+    def _model_endpoint_draft(
+        self,
+        name,
```
Comment (Eyal-Danieli) [inline server/py/services/api/crud/model_monitoring/deployment.py]:
```suggestion
        name: str,
```

File: server/py/services/api/crud/model_monitoring/model_endpoints.py [hunk 5]
```diff
@@ -74,27 +75,64 @@ def create_model_endpoint(
             * **archive**:
             1. If model endpoints with the same name exist, preserve them.
             2. Create a new model endpoint with the same name and set it to `latest`.
+        :param model_path:             The path to the model artifact.
 
-        :return: `ModelEndpoint` object.
+        :return: The crated `ModelEndpoint` object.
```
Comment (Eyal-Danieli) [inline server/py/services/api/crud/model_monitoring/model_endpoints.py]:
```suggestion
        :return: The created `ModelEndpoint` object or `None` if the creation strategy is `SKIP`. 
```

File: server/py/services/api/crud/model_monitoring/model_endpoints.py [hunk 6]
```diff
@@ -52,11 +52,12 @@
...
     """Provide different methods for handling model endpoints such as listing, writing and deleting"""
 
-    def create_model_endpoint(
+    async def create_model_endpoint(
         self,
         db_session: sqlalchemy.orm.Session,
         model_endpoint: mlrun.common.schemas.ModelEndpoint,
         creation_strategy: mlrun.common.schemas.ModelEndpointCreationStrategy,
+        model_path: Optional[str] = None,
     ) -> mlrun.common.schemas.ModelEndpoint:
```
Comment (Eyal-Danieli) [inline server/py/services/api/crud/model_monitoring/model_endpoints.py]:
```suggestion
    ) -> Union[mlrun.common.schemas.ModelEndpoint, None]:
```

File: server/py/services/api/crud/model_monitoring/model_endpoints.py (hunk 5 above)
Comment (Eyal-Danieli) [reply server/py/services/api/crud/model_monitoring/model_endpoints.py]: In addition add 
:raise: MLRunInvalidArgumentError if the creation strategy is not valid

**Review by Eyal-Danieli — COMMENTED** [review]
LGTM, few comments

File: server/py/services/api/crud/model_monitoring/model_endpoints.py (hunk 6 above)
Comment (gtopper) [reply server/py/services/api/crud/model_monitoring/model_endpoints.py]: That would be
```suggestion
    ) -> Optional[mlrun.common.schemas.ModelEndpoint]:
```
But more importantly, all callers need to be ready to handle `None`, and their own type hints need to be adjusted accordingly.

File: server/py/services/api/crud/model_monitoring/deployment.py [hunk 7]
```diff
@@ -1207,6 +1209,253 @@ def _is_the_same_cred(self, stream_path: str, tsdb_connection: str) -> bool:
...
+        return mlrun.common.schemas.ModelEndpoint(
+            metadata=mlrun.common.schemas.ModelEndpointMetadata(
+                project=self.project,
+                name=name,
+                endpoint_type=endpoint_type,
+            ),
+            spec=mlrun.common.schemas.ModelEndpointSpec(
+                function_name=function_name,
+                function_tag=function_tag,
+                function_uid=f"{unversioned_tagged_object_uid_prefix}{function_tag}",  # remove after ML-8596
```
Comment (gtopper) [inline server/py/services/api/crud/model_monitoring/deployment.py]:
```suggestion
                function_uid=f"{unversioned_tagged_object_uid_prefix}{function_tag}",  # TODO: remove after ML-8596
```

**Review by Eyal-Danieli — APPROVED** [review]

**Pull request merged**